spirv_msl.cpp

```diff
@@ -9020,14 +9065,19 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 		if (!msl_options.supports_msl_version(2, 3))
 			SPIRV_CROSS_THROW("discard_fragment() does not formally have demote semantics until MSL 2.3.");
 		CompilerGLSL::emit_instruction(instruction);
+		if (needs_frag_discard_checks())
+			statement(frag_discarded_var_name, " = true;");
 		break;
 
 	case OpIsHelperInvocationEXT:
 		if (msl_options.is_ios() && !msl_options.supports_msl_version(2, 3))
 			SPIRV_CROSS_THROW("simd_is_helper_thread() requires MSL 2.3 on iOS.");
 		else if (msl_options.is_macos() && !msl_options.supports_msl_version(2, 1))
 			SPIRV_CROSS_THROW("simd_is_helper_thread() requires MSL 2.1 on macOS.");
-		emit_op(ops[0], ops[1], "simd_is_helper_thread()", false);
+		if (needs_frag_discard_checks())
```

Comment:
Rather than having two checks here, is it possible to solve this and volatile IsHelperInvocation builtin in one fell swoop, by using reusing the "is helper" builtin here?

BuiltInHelperInvocation is translated to `simd_is_helper_thread()` inline, but if that is lowered to a temporary instead it could unify the IsHelperInvocationEXT and volatile HelperInvocation paths I think.

Reply:
Done.